src/cgns_f.F90

```diff
@@ -5046,6 +5055,331 @@ END SUBROUTINE cgio_set_label_f
+     END IF
+     IF ((ier .EQ. 0) .AND. (PRESENT(index16))) THEN
+        CALL cg_gorel_f1(fn, ier, name16, index16)
+     ELSE
+        RETURN
+     END IF
+     IF ((ier .EQ. 0) .AND. (PRESENT(index17))) THEN
+        CALL cg_gorel_f1(fn, ier, name17, index17)
+     ELSE
+        RETURN
+     END IF
+     IF ((ier .EQ. 0) .AND. (PRESENT(index18))) THEN
+        CALL cg_gorel_f1(fn, ier, name18, index18)
+     ELSE
+        RETURN
+     END IF
+     IF ((ier .EQ. 0) .AND. (PRESENT(index19))) THEN
+        CALL cg_gorel_f1(fn, ier, name19, index19)
+     ELSE
+        RETURN
+     END IF
+  END SUBROUTINE
+
+  SUBROUTINE cg_gorel_f(fn, ier, &
+                  name1, index1,  &
+                  name2, index2,  &
+                  name3, index3,  &
+                  name4, index4,  &
+                  name5, index5,  &
+                  name6, index6,  &
+                  name7, index7,  &
+                  name8, index8,  &
+                  name9, index9,  &
+                  name10, index10,  &
+                  name11, index11,  &
+                  name12, index12,  &
+                  name13, index13,  &
+                  name14, index14,  &
+                  name15, index15,  &
+                  name16, index16,  &
+                  name17, index17,  &
+                  name18, index18,  &
+                  name19, index19,  &
+                  name20)
+     IMPLICIT NONE
+     INTEGER :: fn
+     INTEGER, INTENT(OUT) :: ier
+     CHARACTER(*) :: name1
+     INTEGER, OPTIONAL :: index1
+     CHARACTER(*), OPTIONAL :: name2
+     INTEGER, OPTIONAL :: index2
+     CHARACTER(*), OPTIONAL :: name3
+     INTEGER, OPTIONAL :: index3
+     CHARACTER(*), OPTIONAL :: name4
+     INTEGER, OPTIONAL :: index4
+     CHARACTER(*), OPTIONAL :: name5
+     INTEGER, OPTIONAL :: index5
+     CHARACTER(*), OPTIONAL :: name6
+     INTEGER, OPTIONAL :: index6
+     CHARACTER(*), OPTIONAL :: name7
+     INTEGER, OPTIONAL :: index7
+     CHARACTER(*), OPTIONAL :: name8
+     INTEGER, OPTIONAL :: index8
+     CHARACTER(*), OPTIONAL :: name9
+     INTEGER, OPTIONAL :: index9
+     CHARACTER(*), OPTIONAL :: name10
+     INTEGER, OPTIONAL :: index10
+     CHARACTER(*), OPTIONAL :: name11
+     INTEGER, OPTIONAL :: index11
+     CHARACTER(*), OPTIONAL :: name12
+     INTEGER, OPTIONAL :: index12
+     CHARACTER(*), OPTIONAL :: name13
+     INTEGER, OPTIONAL :: index13
+     CHARACTER(*), OPTIONAL :: name14
+     INTEGER, OPTIONAL :: index14
+     CHARACTER(*), OPTIONAL :: name15
+     INTEGER, OPTIONAL :: index15
+     CHARACTER(*), OPTIONAL :: name16
+     INTEGER, OPTIONAL :: index16
+     CHARACTER(*), OPTIONAL :: name17
+     INTEGER, OPTIONAL :: index17
+     CHARACTER(*), OPTIONAL :: name18
+     INTEGER, OPTIONAL :: index18
+     CHARACTER(*), OPTIONAL :: name19
+     INTEGER, OPTIONAL :: index19
+     CHARACTER(*), OPTIONAL :: name20
+
+     IF (.NOT. PRESENT(index1)) THEN
+        CALL cg_gorel_f1(fn, ier, name1, 0)
+        RETURN
```

Comment:
The issue with the RETURNs is you are going to miss the corner case if a user specifies, for example, name2="x" and index=2, which might be unusual but is allowed by the standard.

Reply:
How is this corner case possible ? 
Indeed if index1 is not present, it should imply that name1=="end" and the goto do not change position. The proposed code is not checking that name1 is indeed equal to "end" and is relying on the fact that underlying C code will reach name1  before the other "end" in cg_gorel_f1 (weird choice of coding??) to stop treating the input parameters.